warehouse/internal/repo/load.go

```diff
@@ -43,6 +45,11 @@ func NewLoadFiles(db *sql.DB, opts ...Opt) *LoadFiles {
 
 // DeleteByStagingFiles deletes load files associated with stagingFileIDs.
 func (repo *LoadFiles) DeleteByStagingFiles(ctx context.Context, stagingFileIDs []int64) error {
+	ctx, cancel := context.WithTimeout(
+		ctx,
+		config.GetDuration("Warehouse.deleteLoadFilesTimeout", 5, time.Minute),
```

Comment:
Having such fine-grain control over the timeouts might make it harder to change the configuration during runtime. Assuming there is a problem with the database that affects all timing, we have to track down all the queries.

 Having such great timeouts I find it hard to require different configs for each of our queries.